task/mgmt-gateway/src/update/sp.rs

```diff
@@ -166,25 +333,29 @@ impl SpUpdate {
 
         match current.state() {
             // Inactive states - nothing to do in response to an abort.
-            State::Complete | State::Aborted => return Ok(()),
-
-            // Active states - fall through to the rest of our function to
-            // actually perform the abort.
-            State::AcceptingData { .. } | State::Failed(_) => (),
-        }
-
-        if *id != current.id() {
-            return Err(ResponseError::UpdateInProgress(self.status()));
-        }
+            State::Complete | State::Aborted => Ok(()),
+
+            // Active states - actually perform the abort.
+            State::AuxFlash(_)
+            | State::FoundMatchingAuxFlashChck { .. }
+            | State::AcceptingData { .. }
+            | State::Failed(_) => {
+                if *id != current.id() {
+                    return Err(ResponseError::UpdateInProgress(self.status()));
+                }
 
-        match self.sp_task.abort_update() {
-            // Aborting an update that hasn't started yet is fine;
-            // either way our caller is clear to start a new update.
-            Ok(()) | Err(UpdateError::UpdateNotStarted) => {
-                *current.state_mut() = State::Aborted;
-                Ok(())
+                match self.sp_task.abort_update() {
```

Comment:
Do we need to call into `self.auxflash_task` instead of `self.sp_task` if we're in an auxflash state?

Reply:
No; there's nothing to abort on the aux flash side, but we do need to tell the SP `UpdateServer` to abort, even if we were still in one of the aux flash states (because we told it to prepare for an update back in `prepare()`).

Reply:
Noted this in 1fc90851